### PR TITLE
Add vim modeline.

### PR DESCRIPTION
**Pull request by dragonwu0919**
I add vim modeline 

https://github.com/solvcon/modmesh/blob/3211479bce05b31a7a483085df6eb7f5a4fa7c94/cpp/modmesh/buffer/SimpleArray.cpp#L519

### CODE REVIEW

**Comment by dragonwu0919** [comment]
By the way, I have found many modeline written in different form. Should these modeline be re-written?

for example, instead of `// vim: set ff=unix fenc=utf8 et sw=4 ts=4 sts=4:`, some the other forms could be:

https://github.com/solvcon/modmesh/blob/3211479bce05b31a7a483085df6eb7f5a4fa7c94/cpp/modmesh/buffer/BufferExpander.hpp#L143

https://github.com/solvcon/modmesh/blob/3211479bce05b31a7a483085df6eb7f5a4fa7c94/cpp/modmesh/buffer/pymod/buffer_pymod.cpp#L64

**Comment by tigercosmos** [comment]
I think we can use the longest version as it includes all other versions' settings. 🤔

**Comment by dragonwu0919** [comment]
@yungyuc I have added one vim modeline. Please help me review it. Thank.

**Comment by ThreeMonth03** [comment]
In [modmesh/gtests/](https://github.com/solvcon/modmesh/tree/master/gtests), it seems that the vim modeline should be added in some of the files. @dragonwu0919, could you please check more files and add more missing vim modelines? Thank you.

**Comment by yungyuc** [comment]
> In [modmesh/gtests/](https://github.com/solvcon/modmesh/tree/master/gtests), it seems that the vim modeline should be added in some of the files. @dragonwu0919, could you please check more files and add more missing vim modelines? Thank you.

I concur.  @dragonwu0919 it will help you get familiar with the code base to scan through all files for a specific style conformity/violation.  Why don't you compile a list of all files that need to fix for the modeline?

**Comment by dragonwu0919** [comment]
@ThreeMonth03 Sure, I can help add the missing vim modeline back.

@yungyuc Thank you for the review and the suggestion. I will try to compile them when fixing modelines.

**Comment by ThreeMonth03** [comment]
> @ThreeMonth03 Sure, I can help add the missing vim modeline back.
> 
> @yungyuc Thank you for the review and the suggestion. I will try to compile them when fixing modelines.

It looks like you've successfully rebased and squashed the commits. @dragonwu0919, could you please remove the "Add Vim modelines 2" commit message because it is meaningless? Thank you.

**Comment by dragonwu0919** [comment]
> > @ThreeMonth03 Sure, I can help add the missing vim modeline back.
> > @yungyuc Thank you for the review and the suggestion. I will try to compile them when fixing modelines.
> 
> It looks like you've successfully rebased and squashed the commits. @dragonwu0919, could you please remove the "Add Vim modelines 2" commit because it is meaningless? Thank you.

@ThreeMonth03 Sorry for asking a basic question, but may i ask how to remove that commit? My log shows that the commit "Add Vim modeline 2" has been deleted in my local repo.
```
PS C:\SciWork\modmesh> git log --oneline
cbfa15a (HEAD -> CodeStyle) Add vim modeline.
f7195c7 (origin/master, origin/HEAD, master) Merge pull request #418 from Tucchhaa/expose_camera_api_python
42161b5 apply review 2
```

**Comment by ThreeMonth03** [comment]
> > > @ThreeMonth03 Sure, I can help add the missing vim modeline back.
> > > @yungyuc Thank you for the review and the suggestion. I will try to compile them when fixing modelines.
> > 
> > 
> > It looks like you've successfully rebased and squashed the commits. @dragonwu0919, could you please remove the "Add Vim modelines 2" commit because it is meaningless? Thank you.
> 
> @ThreeMonth03 Sorry for asking a basic question, but may i ask how to remove that commit? My powershell says that the commit "Add Vim modeline 2" has been deleted in my local repo.
> 
> ```
> PS C:\SciWork\modmesh> git log --oneline
> cbfa15a (HEAD -> CodeStyle) Add vim modeline.
> f7195c7 (origin/master, origin/HEAD, master) Merge pull request #418 from Tucchhaa/expose_camera_api_python
> 42161b5 apply review 2
> ``
> ```

Sorry for my unclear explanation. I think you should rebase and reword the commit `cbfa15a`, and you could refer the note to do it.

**Comment by ThreeMonth03** [comment]
> > > @ThreeMonth03 Sure, I can help add the missing vim modeline back.
> > > @yungyuc Thank you for the review and the suggestion. I will try to compile them when fixing modelines.
> > 
> > 
> > It looks like you've successfully rebased and squashed the commits. @dragonwu0919, could you please remove the "Add Vim modelines 2" commit because it is meaningless? Thank you.
> 
> @ThreeMonth03 Sorry for asking a basic question, but may i ask how to remove that commit? My log shows that the commit "Add Vim modeline 2" has been deleted in my local repo.
> 
> ```
> PS C:\SciWork\modmesh> git log --oneline
> cbfa15a (HEAD -> CodeStyle) Add vim modeline.
> f7195c7 (origin/master, origin/HEAD, master) Merge pull request #418 from Tucchhaa/expose_camera_api_python
> 42161b5 apply review 2
> ```

BTW, if you enter `git log` in your terminal, you should see that the second line of the commit `cbfa15a` is `Add Vim modeline 2`, and you could reword the commit `cbfa15a` and remove this line in the commit message.

**Comment by ThreeMonth03** [comment]
Thanks, @dragonwu0919, the commit message is satisfying.

**Comment by dragonwu0919** [comment]
> > > > @ThreeMonth03 Sure, I can help add the missing vim modeline back.
> > > > @yungyuc Thank you for the review and the suggestion. I will try to compile them when fixing modelines.
> > > 
> > > 
> > > It looks like you've successfully rebased and squashed the commits. @dragonwu0919, could you please remove the "Add Vim modelines 2" commit because it is meaningless? Thank you.
> > 
> > 
> > @ThreeMonth03 Sorry for asking a basic question, but may i ask how to remove that commit? My log shows that the commit "Add Vim modeline 2" has been deleted in my local repo.
> > ```
> > PS C:\SciWork\modmesh> git log --oneline
> > cbfa15a (HEAD -> CodeStyle) Add vim modeline.
> > f7195c7 (origin/master, origin/HEAD, master) Merge pull request #418 from Tucchhaa/expose_camera_api_python
> > 42161b5 apply review 2
> > ```
> 
> BTW, if you enter `git log` in your terminal, you should see that the second line of the commit `cbfa15a` is `Add Vim modeline 2`, and you could reword the commit `cbfa15a` and remove this line in the commit message.

@ThreeMonth03  Thank you for the reminding. I never thought that `--oneline` will function like this. I misunderstood this command into *squish commit log into one line*, instead of *show only the first line of commit log* .

**Comment by dragonwu0919** [comment]
> @ThreeMonth03 Sure, I can help add the missing vim modeline back.
> 
> @yungyuc Thank you for the review and the suggestion. I will try to compile them when fixing modelines.

@yungyuc Sorry for bothering you further. You suggested that I could ***compile*** those files when fixing the modeline problem, but I don't understand what the word ***compile*** means in this context. Should I literally find a compiler to compile the files, or is it just suggesting that I could read the pattern of how the code is written here?

**Comment by yungyuc** [comment]
> @yungyuc Sorry for bothering you further. You suggested that I could _**compile**_ those files when fixing the modeline problem, but I don't understand what the word _**compile**_ means in this context. Should I literally find a compiler to compile the files, or is it just suggesting that I could read the pattern of how the code is written here?

By "compile a list of all files that need to fix for the modeline", I meant to make a list of all files that need to fix for the modeline.

**Comment by dragonwu0919** [comment]
> > @yungyuc Sorry for bothering you further. You suggested that I could _**compile**_ those files when fixing the modeline problem, but I don't understand what the word _**compile**_ means in this context. Should I literally find a compiler to compile the files, or is it just suggesting that I could read the pattern of how the code is written here?
> 
> By "compile a list of all files that need to fix for the modeline", I meant to make a list of all files that need to fix for the modeline.

Understood. I will try my best to do so.

**Comment by dragonwu0919** [comment]
@yungyuc The files that have been updated with Vim modelines are listed below. Please review the overall changes. Thank you.

```json
{
    "WithoutModeline":  [
                       ".\\setup.py",
                       ".\\cpp\\modmesh\\buffer\\pymod\\TypeBroadcast.hpp",
                       ".\\cpp\\modmesh\\inout\\pymod\\wrap_Gmsh.cpp",
                       ".\\cpp\\modmesh\\inout\\pymod\\wrap_Plot3d.cpp",
                       ".\\cpp\\modmesh\\onedim\\Euler1DCore.cpp",
                       ".\\cpp\\modmesh\\onedim\\Euler1DCore.hpp",
                       ".\\cpp\\modmesh\\spacetime\\kernel\\BadEuler1DSolver.cpp",
                       ".\\cpp\\modmesh\\spacetime\\kernel\\BadEuler1DSolver.hpp",
                       ".\\cpp\\modmesh\\testhelper\\pymod\\testbuffer_pymod.cpp",
                       ".\\modmesh\\params.py",
                       ".\\modmesh\\app\\euler1d.py",
                       ".\\tests\\test_gmsh.py"
                   ]
}
```

File: cpp/modmesh/buffer/pymod/TypeBroadcast.hpp [hunk 1]
```diff
@@ -234,3 +234,5 @@ struct TypeBroadcast
 
 } /* end namespace python */
 } /* end namespace modmesh */
+\r\n
```
Comment (yungyuc) [inline cpp/modmesh/buffer/pymod/TypeBroadcast.hpp]: It seems that you have some lines mixed with [dos/windows CF+LF plain-text line ending](https://www.cs.toronto.edu/~krueger/csc209h/tut/line-endings.html).  Please correct them.

You probably did not configure your editor correctly.

**Review by yungyuc — CHANGES_REQUESTED** [review]
> The files that have been updated with Vim modelines are listed below. Please review the overall changes. Thank you.
> 
> ```json
> {
>     "WithoutModeline":  [
>                        ".\\setup.py",
>                        ".\\cpp\\modmesh\\buffer\\pymod\\TypeBroadcast.hpp",
>                        ".\\cpp\\modmesh\\inout\\pymod\\wrap_Gmsh.cpp",
>                        ".\\cpp\\modmesh\\inout\\pymod\\wrap_Plot3d.cpp",
>                        ".\\cpp\\modmesh\\onedim\\Euler1DCore.cpp",
>                        ".\\cpp\\modmesh\\onedim\\Euler1DCore.hpp",
>                        ".\\cpp\\modmesh\\spacetime\\kernel\\BadEuler1DSolver.cpp",
>                        ".\\cpp\\modmesh\\spacetime\\kernel\\BadEuler1DSolver.hpp",
>                        ".\\cpp\\modmesh\\testhelper\\pymod\\testbuffer_pymod.cpp",
>                        ".\\modmesh\\params.py",
>                        ".\\modmesh\\app\\euler1d.py",
>                        ".\\tests\\test_gmsh.py"
>                    ]
> }
> ```

Thanks.  It is clear.  Having said that, it will look nicer in the review by using a markdown list:

* `setup.py`
* `cpp/modmesh/buffer/pymod/TypeBroadcast.hpp`
* `cpp/modmesh/inout/pymod/wrap_Gmsh.cpp`
* `cpp/modmesh/inout/pymod/wrap_Plot3d.cpp`
* `cpp/modmesh/onedim/Euler1DCore.cpp`
* `cpp/modmesh/onedim/Euler1DCore.hpp`
* `cpp/modmesh/spacetime/kernel/BadEuler1DSolver.cpp`
* `cpp/modmesh/spacetime/kernel/BadEuler1DSolver.hpp`
* `cpp/modmesh/testhelper/pymod/testbuffer_pymod.cpp`
* `modmesh/params.py`
* `modmesh/app/euler1d.py`
* `tests/test_gmsh.py`

(Simple regular expressions can convert the json to the markdown list.)

---

Need to change:

- [ ] Make sure all line endings uses Unix LF (`\n`).

File: cpp/modmesh/spacetime/kernel/BadEuler1DSolver.hpp [hunk 2]
```diff
@@ -300,4 +300,6 @@ inline void BadEuler1DSolver::march_alpha(size_t steps)
 }
 
 } /* end namespace spacetime */
-} /* end namespace modmesh */
\ No newline at end of file
+} /* end namespace modmesh */
+// vim: set ff=unix fenc=utf8 et sw=4 ts=4 sts=4:
```
Comment (ThreeMonth03) [inline cpp/modmesh/spacetime/kernel/BadEuler1DSolver.hpp]: I'm not sure whether there should be a newline between the bracket and the vim modeline. @yungyuc, how do you think?

File: gtests/test_nopython_callprofiler.cpp [hunk 3]
```diff
@@ -213,4 +213,6 @@ TEST_F(CallProfilerTest, cancel)
 }
 
 } // namespace detail
-} // namespace modmesh
\ No newline at end of file
+} // namespace modmesh
+
+// vim: set ff=unix fenc=utf8 et sw=4 ts=4 sts=4:
```
Comment (ThreeMonth03) [inline gtests/test_nopython_callprofiler.cpp]: Missing newline (\n).

File: modmesh/app/euler1d.py [hunk 4]
```diff
@@ -839,3 +839,6 @@ def load_app():
     config_window.redraw()
     plotting_area.redraw()
     _subwin.show()
+
+# vim: set ff=unix fenc=utf8 et sw=4 ts=4 sts=4: 
+
```
Comment (ThreeMonth03) [inline modmesh/app/euler1d.py]: image

In the editor, it is clear that there are two newlines after the vim modeline, and the github action should fail under this circumstance.
Thus, you may modify this part according to the following rules:
1. There should be only a newline after the vim modeline.
2. The newline should not contain any character, such as the tab `'\t'` and the space ` `.

File: cpp/modmesh/buffer/pymod/TypeBroadcast.hpp [hunk 5]
```diff
@@ -234,3 +234,6 @@ struct TypeBroadcast
 
 } /* end namespace python */
 } /* end namespace modmesh */
+
+// vim: set ff=unix fenc=utf8 et sw=4 ts=4 sts=4:
+
```
Comment (ThreeMonth03) [inline cpp/modmesh/buffer/pymod/TypeBroadcast.hpp]: 2 newlines.

File: cpp/modmesh/inout/pymod/wrap_Gmsh.cpp [hunk 6]
```diff
@@ -44,3 +44,6 @@ void wrap_Gmsh(pybind11::module & mod)
 } /* end namespace python */
 
 } /* end namespace modmesh */
+
+// vim: set ff=unix fenc=utf8 et sw=4 ts=4 sts=4:
+
```
Comment (ThreeMonth03) [inline cpp/modmesh/inout/pymod/wrap_Gmsh.cpp]: 2 newlines.

File: cpp/modmesh/inout/pymod/wrap_Plot3d.cpp [hunk 7]
```diff
@@ -44,3 +44,6 @@ void wrap_Plot3d(pybind11::module & mod)
 } /* end namespace python */
 
 } /* end namespace modmesh */
+
+// vim: set ff=unix fenc=utf8 et sw=4 ts=4 sts=4:
+
```
Comment (ThreeMonth03) [inline cpp/modmesh/inout/pymod/wrap_Plot3d.cpp]: 2 newlines.

File: cpp/modmesh/onedim/Euler1DCore.cpp [hunk 8]
```diff
@@ -228,3 +228,6 @@ SimpleArray<double> Euler1DCore::entropy() const
 
 } /* end namespace onedim */
 } /* end namespace modmesh */
+
+// vim: set ff=unix fenc=utf8 et sw=4 ts=4 sts=4:
+
```
Comment (ThreeMonth03) [inline cpp/modmesh/onedim/Euler1DCore.cpp]: 2 newlines.

File: cpp/modmesh/onedim/Euler1DCore.hpp [hunk 9]
```diff
@@ -344,4 +344,6 @@ inline void Euler1DCore::march_alpha(size_t steps)
 }
 
 } /* end namespace onedim */
-} /* end namespace modmesh */
\ No newline at end of file
+} /* end namespace modmesh */
+// vim: set ff=unix fenc=utf8 et sw=4 ts=4 sts=4:
+
```
Comment (ThreeMonth03) [inline cpp/modmesh/onedim/Euler1DCore.hpp]: 2 newlines.

File: cpp/modmesh/spacetime/kernel/BadEuler1DSolver.cpp [hunk 10]
```diff
@@ -131,3 +131,6 @@ void BadEuler1DSolver::treat_boundary_so1()
 
 } /* end namespace spacetime */
 } /* end namespace modmesh */
+
+// vim: set ff=unix fenc=utf8 et sw=4 ts=4 sts=4:
+
```
Comment (ThreeMonth03) [inline cpp/modmesh/spacetime/kernel/BadEuler1DSolver.cpp]: 2 newlines.

File: cpp/modmesh/testhelper/pymod/testbuffer_pymod.cpp [hunk 11]
```diff
@@ -61,3 +61,6 @@ void initialize_testbuffer(pybind11::module & mod)
 } /* end namespace python */
 
 } /* end namespace modmesh */
+
+// vim: set ff=unix fenc=utf8 et sw=4 ts=4 sts=4:
+
```
Comment (ThreeMonth03) [inline cpp/modmesh/testhelper/pymod/testbuffer_pymod.cpp]: 2 newlines.

File: gtests/test_nopython_inout.cpp [hunk 12]
```diff
@@ -314,3 +314,5 @@ TEST(Gmsh_Parser, Hexahedron125NodeDefinition)
     EXPECT_EQ(ele_def.mmtpn(), 5);
     EXPECT_THAT(ele_def.mmcl(), testing::ElementsAre(0, 1, 2, 3, 4, 5, 6, 7));
 }
+
+// vim: set ff=unix fenc=utf8 et sw=4 ts=4 sts=4:
```
Comment (ThreeMonth03) [inline gtests/test_nopython_inout.cpp]: missing newline.

File: gtests/test_nopython_serializable.cpp [hunk 13]
```diff
@@ -322,4 +322,6 @@ TEST(Json, deserialize_with_unordered_map)
     EXPECT_EQ(item.pet_map["cat"].is_cat, true);
 }
 
-} // namespace modmesh
\ No newline at end of file
+} // namespace modmesh
+
+// vim: set ff=unix fenc=utf8 et sw=4 ts=4 sts=4:
```
Comment (ThreeMonth03) [inline gtests/test_nopython_serializable.cpp]: missing newline.

File: modmesh/params.py [hunk 14]
```diff
@@ -109,3 +109,6 @@ def run(self):
 
     root = Example()
     root.run()
+
+# vim: set ff=unix fenc=utf8 et sw=4 ts=4 sts=4: 
+
```
Comment (ThreeMonth03) [inline modmesh/params.py]: 2 newlines.

File: setup.py [hunk 15]
```diff
@@ -79,3 +79,6 @@ def main():
 
 if __name__ == '__main__':
     main()
+
+# vim: set ff=unix fenc=utf8 et sw=4 ts=4 sts=4: 
+
```
Comment (ThreeMonth03) [inline setup.py]: 2 newlines.

File: tests/test_gmsh.py [hunk 16]
```diff
@@ -33,3 +33,6 @@ def test_gmsh_parsing(self):
         self.assertEqual(blk.clnds.ndarray[3:, :4].tolist(), [[3, 0, 1, 2],
                                                               [3, 0, 2, 3],
                                                               [3, 0, 3, 1]])
+
+# vim: set ff=unix fenc=utf8 et sw=4 ts=4 sts=4: 
+
```
Comment (ThreeMonth03) [inline tests/test_gmsh.py]: 2 newlines.

File: cpp/modmesh/buffer/SimpleArray.cpp [hunk 17]
```diff
@@ -515,3 +515,5 @@ SimpleArrayPlex::~SimpleArrayPlex()
 }
 
 } /* end namespace modmesh */
+
+// vim: set ff=unix fenc=utf8 et sw=4 ts=4 sts=4:
```
Comment (ThreeMonth03) [inline cpp/modmesh/buffer/SimpleArray.cpp:519]: There is only one newline after the vim modeline, good job.

**Review by ThreeMonth03 — COMMENTED** [review]
- [ ] Keep only one newline after the vim modeline.
You may refer [cpp/modmesh/buffer/SimpleArray.cpp](https://github.com/solvcon/modmesh/pull/420/files/33734c6a50b870086706167e41bddc8c9d5c974c#diff-04216ffd67af1f4755bacaae487e30f1bf519d8302cf86fe7864fe08b8182d62) and [this explanation](https://github.com/solvcon/modmesh/pull/420#discussion_r1739892172) to modify the files.

File: cpp/modmesh/spacetime/kernel/BadEuler1DSolver.hpp (hunk 2 above)
Comment (yungyuc) [reply cpp/modmesh/spacetime/kernel/BadEuler1DSolver.hpp]: It is better to leave a blank line between the modeline and the code above it.

**Comment by dragonwu0919** [comment]
- [x] Ensured there is only one newline after the modeline.
- [x] Ensured there is a newline between the Vim modeline and the rest of the file content.

Hope it doesn't go wrong this time. @yungyuc, could you please help me review the changes? Thank you!

File: gtests/test_nopython_callprofiler.cpp (hunk 3 above)
Comment (ThreeMonth03) [reply gtests/test_nopython_callprofiler.cpp]: image

It seems that you miss to add a newline after the vim modeline.

File: gtests/test_nopython_inout.cpp (hunk 12 above)
Comment (ThreeMonth03) [reply gtests/test_nopython_inout.cpp]: It seems that you miss to add a newline after the vim modeline.

image

File: gtests/test_nopython_serializable.cpp (hunk 13 above)
Comment (ThreeMonth03) [reply gtests/test_nopython_serializable.cpp]: It seems that you miss to add a newline after the vim modeline.

image

**Review by ThreeMonth03 — CHANGES_REQUESTED** [review]
- [ ] Add the missing newline in [gtests/test_nopython_callprofiler.cpp](https://github.com/solvcon/modmesh/pull/420/files#diff-f2b5f43898d27aa295793363d34b89b0ef47c3beacb2ed3dd2756797f28f044c), [gtests/test_nopython_inout.cpp](https://github.com/solvcon/modmesh/pull/420/files#diff-da7d5e51ea17281d289ead2dc96f790f910914b3492407efc1c3223ba65bf08e) and [gtests/test_nopython_serializable.cpp](https://github.com/solvcon/modmesh/pull/420/files#diff-44ff83115bf707eb36686b238dc2a5a150058e3caa924025a68a164819e66163).

@dragonwu0919 I think after the slight modification to the above files, we could merge this pr. Thank's for your contribution.

**Comment by yungyuc** [comment]
> * [ ]  Add the missing newline in [gtests/test_nopython_callprofiler.cpp](https://github.com/solvcon/modmesh/pull/420/files#diff-f2b5f43898d27aa295793363d34b89b0ef47c3beacb2ed3dd2756797f28f044c), [gtests/test_nopython_inout.cpp](https://github.com/solvcon/modmesh/pull/420/files#diff-da7d5e51ea17281d289ead2dc96f790f910914b3492407efc1c3223ba65bf08e) and [gtests/test_nopython_serializable.cpp](https://github.com/solvcon/modmesh/pull/420/files#diff-44ff83115bf707eb36686b238dc2a5a150058e3caa924025a68a164819e66163).
> 
> @dragonwu0919 I think after the slight modification to the above files, we could merge this pr. Thank's for your contribution.

Agree.

---

It would be nice to have consistent new line after the modeline, but this is indeed very easy to miss when we only review a limited number of source files.

When it is for developing a feature or fixing a bug, the inconsistency at the end of file is fine.  Picking a minor issue like that after a great effort of hacking can be discouraging.

But for a (style) refactoring PR like this, it good to make the style as consistent as possible, because its sole purpose is to make the style perfect.

**Comment by dragonwu0919** [comment]
I have completed the modifications for these three files. @yungyuc, @ThreeMonth03, thank you both very much. This experience has really helped me become more familiar with how PRs work.

**Review by ThreeMonth03 — APPROVED** [review]
@dragonwu0919 Thanks for your contribution. Everything looks fine now.

**Comment by ThreeMonth03** [comment]
> @dragonwu0919 Thanks for your contribution. Everything looks fine now.

@dragonwu0919 We are sorry to inform you that you haven't pass all of the github workflows, and the detail of the error log could find [here](https://github.com/solvcon/modmesh/actions/runs/10652443284/job/29527238961?pr=420#step:13:31). You must to pass all of the the github action before merging the pr.
image

BTW, if you want to run the github action under your repository, you can refer [this article](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository), then your repository will run the [github workflows](https://github.com/dragonwu0919/modmesh/actions) automatically.

**Comment by ThreeMonth03** [comment]
> > @dragonwu0919 Thanks for your contribution. Everything looks fine now.
> 
> @dragonwu0919 We are sorry to inform you that you haven't pass all of the github workflows, and the detail of the error log could find [here](https://github.com/solvcon/modmesh/actions/runs/10652443284/job/29527238961?pr=420#step:13:31). You must to pass all of the the github action before merging the pr. image
> 
> BTW, if you want to run the github action under your repository, you can refer [this article](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository), then your repository will run the [github workflows](https://github.com/dragonwu0919/modmesh/actions) automatically.

@dragonwu0919, thank you very much. It seems that you correct the errors and pass [the workflows](https://github.com/dragonwu0919/modmesh/actions), and it looks good to me. @yungyuc, you could review this pr when you are available, thank you.

**Review by yungyuc — APPROVED** [review]

**Comment by yungyuc** [comment]
Thanks, @dragonwu0919 .  Look good to me.  Merging.

**Pull request merged**